Content.Server/Doors/Systems/FirelockSystem.cs

```diff
@@ -51,36 +43,13 @@ public override void Update(float frameTime)
 
             _accumulatedTicks = 0;
 
-            var airtightQuery = GetEntityQuery<AirtightComponent>();
-            var appearanceQuery = GetEntityQuery<AppearanceComponent>();
-            var xformQuery = GetEntityQuery<TransformComponent>();
-            var pointLightQuery = GetEntityQuery<PointLightComponent>();
-
+            // Go through all of the firelocks and re-close them if they've been pried open.
             var query = EntityQueryEnumerator<FirelockComponent, DoorComponent>();
             while (query.MoveNext(out var uid, out var firelock, out var door))
             {
-                // only bother to check pressure on doors that are some variation of closed.
-                if (door.State != DoorState.Closed
-                    && door.State != DoorState.Welded
-                    && door.State != DoorState.Denying)
+                if (firelock.IsLocked && _gameTiming.CurTime > firelock.EmergencyCloseCooldown)
                 {
-                    continue;
-                }
-
-                if (airtightQuery.TryGetComponent(uid, out var airtight)
-                    && xformQuery.TryGetComponent(uid, out var xform)
-                    && appearanceQuery.TryGetComponent(uid, out var appearance))
-                {
-                    var (pressure, fire) = CheckPressureAndFire(uid, firelock, xform, airtight, airtightQuery);
```

Comment:
Put pressure checks back, @ShadowCommander pointed out a very good use case for having pressure differential stored.